File: Level3/정수 삼각형.py
def solution(triangle):
    answer = []
    for i in range(len(triangle)-1):
        new = [0] * (i+2)
        for p in range(len(triangle[i])):
            a = triangle[i][p]
            for j in range(p, p+2):
                b = triangle[i+1][j]
                new[j] = max(new[j], a+b)
        triangle[i+1] = new
    return max(triangle[-1])

File: Level3/정수 삼각형.java
import java.util.*;
class Solution {
    public int solution(int[][] triangle) {
        int answer = 0;
        int[][] test = new int[triangle.length][triangle.length];
        test[0][0] = triangle[0][0];
        for(int i=1; i<triangle.length; i++) {
            for(int j=0; j<=i; j++) {
                if(j == 0) 
                    test[i][j] = triangle[i][j] + test[i-1][j];
                else if(j == i)
                    test[i][j] = triangle[i][j] + test[i-1][j-1];
                else 
                    test[i][j] = triangle[i][j] + Math.max(test[i-1][j-1], test[i-1][j]);
            }
        }
        
        for(int i=0; i<triangle.length; i++)
            answer = Math.max(answer, test[triangle.length-1][i]);
        
        return answer;
    }
}
